### DISCUSSION TITLE
stash.loadData()

### DISCUSSION THREAD

**Discussion by mailtwogopal** [pactumjs/pactum#188] [Q&A] [answered]
I have my data stored in data/templates folder as .json. And in the main test file, I want to use the data inside .withBody() for a POST request. I know syntax to load data is stash.loadData('path/to/file'). When I directly use like this below, my data is not loaded

```js
await spec().withPath('urlofendpint').withHeaders(headerinfo).withBody(stash.loadData('path/to/file')).expecStatus(200)
```
` => This is not working`

I know for stash.addDataTemplate => We give unique name and we call as @DATA:TEMPLATE@ : nameoftemplate. Can you help for loadData scenario as mentioned in example line (in bold above)

**Comment by ASaiAnudeep** [comment, marked as answer]
The `loadData` function will read the JSON files in the specified directories and loads data templates and maps.

```js
const { stash } = require('pactum');

stash.loadData('/path/to/data/folder');
```

Once the data templates and maps are loaded, we can directly refer them using special syntax. The overall code looks like:

```js
const { stash, spec } = require('pactum');

stash.loadData('/path/to/data/folder');

await spec()
  .post('https://httpbin.org/anything')
  .withJson({
    '@DATA:TEMPLATE@': 'User'
  });
```

**Comment by mailtwogopal** [comment]
Hi Anudeep,

I tried exactly as in email I received but in the log I see req body loaded as is like '@DATA:TEMPLATE@' : 'user'. This is happening only for loadData but addDataTemplate is working fine.

Again I have that json stored in data/template/user.json. And loading data looks like this

stash.loadData('./data/template/');

I tried to parse the data and it is printing same '@DATA:TEMPLATE@' : 'user' .

Any further help is appreciated. Thank you.

**Reply by ASaiAnudeep** [reply]
seems there is a typo - `'@DaTa:TEMPLATE@' : 'user'`

It should be - `'@DATA:TEMPLATE@': 'user'`

And also make sure your file name ends with `.template.json`

**Reply by mailtwogopal** [reply]
Hi,

This seems to be a GitHub comments issue showing like that instead of all full caps. In my test, it spells correctly like this : 
![image](https://user-images.githubusercontent.com/44076731/181808006-6cadaf3f-1818-43e5-b0fc-1f8e4c73cf68.png)

**Reply by mailtwogopal** [reply]
Thank you, It worked fine after changing data file name to end with ".template.json"
